pandas/core/apply.py

```diff
@@ -837,6 +837,43 @@ def agg_or_apply_list_like(
         if getattr(obj, "axis", 0) == 1:
             raise NotImplementedError("axis other than 0 is not supported")
 
+        # GH#49352 - Handle numeric_only with list of functions
+        # When numeric_only=True is passed with a list of functions, filter
+        # to numeric columns before processing to avoid TypeError on non-numeric Series
+        if op_name == "agg" and kwargs.get("numeric_only", False):
```

Comment:
Only two of the five paths in `DataFrameGroupBy.aggregate` hits here, thus introducing inconsistencies. To accept a PR, we would need to do this consistently through the op. Take a look at the code in `pandas.core.groupby.generic`.